bin/sync_ap_tier1_projects.py

```diff
@@ -0,0 +1,137 @@
+    members: comma-separated list of VSC IDs of project members
+
+    """
+
+    projects_config = ConfigParser()
+    with open(projects_ini) as pini:
+        projects_config.read_file(pini)
+
+    projects = []
+    other_sources = []
+
+    for section in projects_config.sections():
+        if section in VSC_ADMIN_GROUPS:
+            other_sources.append(section)
+
+        if section.startswith('gpr_compute'):
+            projects.append(ProjectIniConfig(
+                name=section,
+                description=projects_config.get(section, "description", fallback=section),
+                end_date=projects_config.get(section, "end_date"),
+                members=set([m.strip() for m in projects_config.get(section, "members").split(",")]),
+                moderators=set([m.strip() for m in projects_config.get(section, "moderators").split(",")]),
+            ))
+
+    return (projects, other_sources)
+
+class Tier1APProjectSync(Sync):
+    CLI_OPTIONS = {
+        'project_ini': ('Ini file with projects information', str, 'store', None),
+    }
+
+
+    def do(self, dryrun):
+
+        # TODO: take end_date into account
+
+        (projects, other_sources) = get_projects(self.options.project_ini)
+        projects_members = [(set(p.members), p.name) for p in projects]  # TODO: verify enddates
+
+        logging.debug("Current projects: %s", [p.name for p in projects])
+        logging.debug("Current other sources: %s", other_sources)
+
+        # get all the groups that corresponds to projects
+        active_groups, _ = self.get_groups(modified_since="20211101")
+        active_group_names = set([g.vsc_id for g in active_groups])
+        active_accounts, inactive_accounts = self.get_accounts()
+
+        active_users_autogroup_sources = self.apc.autogroup[AP_ACTIVE_USERS_AUTOGROUP].get()[1]["sources"]
+
+        logging.debug("Current %s sources: %s", AP_ACTIVE_USERS_AUTOGROUP, active_users_autogroup_sources)
+
+        # create the projects groups in the AP
+        for source in other_sources:
+            if source not in active_users_autogroup_sources:
+                if dryrun:
+                    logging.info("Calling apc.autogroup[%s].source[%s].add.post()", AP_ACTIVE_USERS_AUTOGROUP, source)
+                else:
+                    self.apc.autogroup[AP_ACTIVE_USERS_AUTOGROUP].source[source].add.post()
+
+        for project in projects:
+            if project.name not in active_group_names:
+                data = {
+                    "name": project.name.replace("gpr_compute_", ""),  # this will be regenerated, the RA should only have the suffix
+                    "members": [
+                        { "vsc_id": m, "moderator": m in project.moderators }
+                        for m in project.members.union(project.moderators)
+                    ],
+                    "description": project.description,
+                    "metadata": {
+                        "institute": "gent",
```

Comment:
is this supposed to be fixed?

Reply:
It'll depend on how we decide to assign institute in projects in the resource tracking app. But I'd prefer it to be really per institute.